### DISCUSSION TITLE
[GrapheneOS] Unable to take pictures with Camera on Android

### DISCUSSION THREAD

**Discussion by fsover2** [gsantner/markor#2007] [Q&A] [answered]
I used to be able to insert an image using my phone camera. However, now when I press the "Camera Picture" button, nothing happens. I'm not sure if this has something to do with Markor, or if Android changed something that is now blocking my camera. (I'm running CalyxOS)

I did notice that Markor doesn't request any permissions to access the camera. [See Android permissions.](https://github.com/gsantner/markor#android-permissions) Is this feature still supported? Does anyone have suggestions on how I could get this behavior working again?

Thanks so much for the great software and any help anyone could provide!

**Comment by gsantner** [comment, marked as answer]
> I did notice that Markor doesn't request any permissions to access the camera

Markor doesn't have any camera functionality on it's own,the camera isn't directly accessible. However it can ask other apps to make pictures, so a camera app, and report back what picture was created.

So do you have any Camera app installed? If not, try OpenCamera.

**Reply by fsover2** [reply]
Thanks for the quick response.

At the time of my question, I had two camera apps installed on my CalyxOS phone: AOSP Camera [default], and Google Camera. After your suggestion, I also installed OpenCamera. None of these apps are launching when I use the "Camera Picture" button.

Most likely this is an issue with how CalyxOS is responding to Markor's request to talk to the default camera app. Could you please describe the mechanism Markor uses to talk to the camera app so I can talk with the CalyxOS community to resolve my issue? Thanks!

**Reply by gsantner** [reply]
@fsover2 Intent MediaStore.ACTION_IMAGE_CAPTURE

My Android device is on Android 11 Lineageos.

**Comment by fsover2** [comment]
Thanks for the answer and great app!

**Comment by harshad1** [comment]
Note that I am in the process of making several improvements / fixes for attachments and captures (camera, recordings etc) here https://github.com/gsantner/markor/pull/2106

If one of you is able to test this branch on LineageOS and / or GrapheneOS that would be great. You can download an APK from https://github.com/gsantner/markor/actions/runs/6388836921.

I am almost done. Need to make a few tweaks to keyboard behavior.
